Sources/NetworkProtection/Models/VPNServerSelectionSanitizer.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  VPNServerSelectionSanitizer.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+public enum VPNServerSelectionError: Error {
+    case countryNotFound
+    case fetchingLocationsFailed(Error)
+}
+
+public protocol VPNServerSelectionSanitizing {
+    func sanitizedServerSelectionMethod() async -> NetworkProtectionServerSelectionMethod
+}
+
+public final class VPNServerSelectionSanitizer: VPNServerSelectionSanitizing {
```

Comment:
A few comments here:

1. Let's add unit tests to cover this class in its entirety.
2. The name of this class sounds off to me.  This does more than just sanitize the selection - it effectively manages the entire selection logic (locations vs servers in settings).  Additionally when referencing this in the init parameter below (for `PacketTunnelProvider`) the instance is named `serverSelection`, which makes me think you agree.  Why not name this something more generic like `VPNServerSelector` or `VPNServerSelectionResolver`.
3. To further expand on point 2, I really think that to the `PacketTunnelProvider` whether this does sanitization or not is an implementation detail - what matters is that this class provides a server / location to connect to.

Reply:
Yeah, the naming here gave me a lot of trouble. So just decided to open it up to get opinions. Thanks for the input.